lib/ui/common/ContextualAppBar.dart

```diff
@@ -42,30 +68,36 @@ class ContextualAppBar extends StatelessWidget {
       actions.add(more);
     }
     return AppBar(
-      leading: isRoot ? _buildDismissIcon(context) : _buildBackIcon(),
+      leading: isRoot ? _buildDismissIcon(context) : _buildBackIcon(context),
       backgroundColor: theme.primaryColor,
       elevation: _style.elevation,
       actions: actions,
     );
   }
 
-  Widget _buildBackIcon() {
-    return Container(
-      margin: EdgeInsets.only(left:_style.insets.left),
-      child: BackButton(
+  Widget _buildBackIcon(BuildContext context) {
+    if (_isModal) {
+      return _buildDismissIcon(context);
+    } else {
+      return Container(
+        margin: EdgeInsets.only(left: _style.insets.left),
+        child: BackButton(
```

Comment:
could you check this uses the FarmSmart back icon from the designs?

Reply:
is not using the back icon from the designs, I will change it on this PR

Reply:
DONE